lib/hocon/impl/config_concatenation.rb

```diff
@@ -186,34 +212,34 @@ def resolve_substitutions(context, source)
     end
   end
 
-  def initialize(origin, pieces)
-    super(origin)
-    @pieces = pieces
-
-    if pieces.size < 2
-      raise ConfigBugOrBrokenError, "Created concatenation with less than 2 items: #{self}"
-    end
+  def resolve_status
+    ResolveStatus::UNRESOLVED
+  end
 
-    had_unmergeable = false
-    pieces.each do |p|
-      if p.is_a?(Hocon::Impl::ConfigConcatenation)
-        raise ConfigBugOrBrokenError, "ConfigConcatenation should never be nested: #{self}"
-      end
-      if p.is_a?(Unmergeable)
-        had_unmergeable = true
-      end
+  def replace_child(child, replacement)
+    new_pieces = replace_child_in_list(@pieces, child, replacement)
+    if new_pieces == nil
```

Comment:
definitely can, but I've been writing these conditions like this to match the Java code as closely as possible.